test/bntest.c

```diff
@@ -82,6 +82,7 @@
 #include <openssl/err.h>
 
 #include "../crypto/bn/bn_lcl.h"
+#include "../crypto/bn/bn_prime.h"
```

Comment:
Nothing wrong with unit tests per se, but the test programs in `test/` aren't set up to be unit testers, as they are built as external programs with access to the public API only. One could say that you cheat, and admittedly successfully so, until OpenSSL is configured to build with shared libraries (which the tests are linked with in that case) 

Unless I have missed something, we don't have a unit testing framework for libcrypto yet. There is some unit testing stuff in libssl that I haven't looked at yet. 

Give us a bit of time and I'll see what we can figure out.